documentation/stackoverflow.workspace.adoc

```diff
@@ -218,44 +343,53 @@ RETURN t.name as tag, count(q) AS questions
 ORDER BY questions DESC LIMIT 10;
 ----
 
-== How are tags related to other tags?
+=== User Engagement
 
-Tag correlations:
+By aggregating when users interacted with StakeOverflow, you can see how active they are.
+
+For example, you can aggregate the number of questions answered by month.
 
 [source,cypher]
 ----
-MATCH (t1:Tag)<-[:TAGGED]-()-[:TAGGED]->(t2:Tag)
-WHERE id(t1) < id(t2) and t1.name <> 'neo4j' and t2.name <> 'neo4j'
-RETURN t1.name, t2.name,count(*) as freq
-ORDER BY freq desc LIMIT 10;
+MATCH (u:User)-[:PROVIDED]->()-[:ANSWERED]->
+      (q:Question)-[:TAGGED]->(t:Tag)
+WHERE u.display_name = "cybersam"
+RETURN apoc.date.format(q.creation_date,'s','yyyy-MM') as month, 
+       count(distinct q) as count, collect(distinct t.name) as tags
+ORDER BY month asc
 ----
 
-== Virtual Graphs: Tags
+=== How are tags related to other tags?
+
+Understanding how users tag questions can help you see patterns.
 
-Project tags via co-occurrence with virtual relationships.
+For example, how tags correlate to each other:
 
 [source,cypher]
 ----
 MATCH (t1:Tag)<-[:TAGGED]-()-[:TAGGED]->(t2:Tag)
 WHERE id(t1) < id(t2) and t1.name <> 'neo4j' and t2.name <> 'neo4j'
-WITH t1, t2,count(*) as freq  where freq > 3
-RETURN t1,t2, apoc.create.vRelationship(t1,'OCCURRED',{freq:freq},t2) as rel
+RETURN t1.name, t2.name,count(*) as freq
+ORDER BY freq desc LIMIT 10;
 ----
 
-If you want to, you can also materialize those relationships in the graph and then explore the data in the _Explore_ tab.
+=== Virtual Graphs: Tags
+
+Using virtual relationships can help you find unknown relationships in the graph. 
+
+For example, you can create a virtual relationship between tags that occur together more than three times:
 
 [source,cypher]
 ----
 MATCH (t1:Tag)<-[:TAGGED]-()-[:TAGGED]->(t2:Tag)
-WHERE id(t1) < id(t2) and t1.name <> 'neo4j' and t2.name <> 'neo4j'
-WITH t1, t2,count(*) as freq  where freq > 3
-MERGE (t1)-[r:OCCURRED]-(t2) SET r.freq=freq
-RETURN count(*)
+WHERE id(t1) < id(t2) 
+WITH t1, t2, count(*) as freq WHERE freq > 3
+RETURN t1,t2, apoc.create.vRelationship(t1,'OCCURRED',{freq:freq},t2) AS rel
```

Comment:
I think we should rather keep the rels materializes, so the rels stay in the graph and people can query them and explore them with Bloom etc.

we haven't introduced apoc yet and it adds another level of complexity.

which reminds me perhaps we should do an APOC GA course (https://dev.neo4j.com/apoc-videos)

Reply:
My first instinct was to remove these 2 queries as I thought they added too much complexity. Given you comments I think I will drop them.